### DISCUSSION TITLE
Include detekt-gradle-plugin directly

### DISCUSSION THREAD

**Discussion by kvn-stgl** [detekt/detekt#4712] [Q&A] [answered]
Currently, I'm trying to include the detekt-gradle-plugin (and with this the complete detekt project) directly from the source code to do some deeper debugging research. 
So in fact I have two different projects:
- ~/mainProject
- ~/detekt

How can I include the detekt/detekt-gradle-plugin into my mainProject? Does anybody have a valid example here?

**Comment by kvn-stgl** [comment, marked as answer]
Well.. This was a way easier as I thought. Simply write ```includeBuild("../detekt")``` into the settings.gradle.kts of the mainProject and everything works like a charm 😄

**Comment by TWiStErRob** [comment]
If you don't need to change things, I found it simplest to add a dependency on the libs. This works with anything:

- either in buildSrc/build.gradle (if you have one)
- or in any module's build.gradle

```
// For browsing code only.
testRuntimeOnly("io.gitlab.arturbosch.detekt:detekt-gradle-plugin")
// CLI has to be runtimeOnly, otherwise the transitive runtime dependencies don't resolve.
testRuntimeOnly("io.gitlab.arturbosch.detekt:detekt-cli")
```

This will pull in the libraries as External Libraries and attach all the -sources.jar code.
After indexing you have "Scopes=All Places" filter and you can search and navigate freely.
